File: DSA(C)/queue.c
#include <stdio.h>
#include <conio.h>

void main()
{
    int size = 10;
    int queue[size];
    int FRONT = 0, REAR = -1;
    void enqueue(int a)
    {
        if (REAR == size - 1)
        {
            printf("Overflow\n");
        }
        else
        {
            REAR = REAR + 1;
            queue[REAR] = a;
        }
    };
    void dequeue()
    {
        if (FRONT == -1)
        {
            printf("Underflow\n");
        }
        else
        {
            printf("-> %d\n", queue[FRONT]);
            FRONT = FRONT + 1;
        }
    };
    void display()
    {
        for (int i = FRONT; i <= REAR; i++)
        {
            printf("%d, ", queue[i]);
        }
        printf("\n");
    }
    enqueue(1);
    enqueue(2);
    enqueue(3);
    enqueue(4);
    enqueue(5);
    enqueue(6);
    enqueue(7);
    enqueue(8);
    enqueue(9);
    enqueue(10);
    display();
    enqueue(11);
    dequeue();
    dequeue();
    dequeue();
    dequeue();
    dequeue();
    dequeue();
    dequeue();
    display();
};

File: Sem-5/Java/1_stack_queues/src/Queue.java
public class Queue {
    private final int[] queue;
    private final int size;
    private int front;
    private int rear;

    public Queue(int size) {
        this.size = size;
        this.queue = new int[size];
        this.front = 0;
        this.rear = -1;
    }

    public void enqueue(int value) {
        if (this.rear == this.size - 1) {
            System.out.println("Overflow");
        } else {
            queue[++rear] = value;
        }
    }

    public void dequeue() {
        if (front > rear) {
            System.out.println("Underflow");
        }else {
            System.out.println("Popped: "+queue[++front]);
        }
    }

    public void display(){
        
        int drear = rear;
        int dfront = front;

        System.out.printf("Queue: ");
        for (int i = dfront; i < drear; i++) {
            System.out.printf("%d ", queue[i]);
        }
        System.out.println();
    }
}
